k-distribution/include/builtin/domains.k

```diff
@@ -590,19 +589,31 @@ module ID-PROGRAM-PARSING
               | #UpperId                                             [notInRules, token, autoReject]
 endmodule
 
-module ID
-  imports STRING
+module ID-SYNTAX
+  imports INT-SYNTAX
+  imports STRING-SYNTAX
 
-  syntax Id
+  syntax Id [token]
 
   syntax String ::= Id2String ( Id )    [function, hook(STRING.token2string)]
-  syntax Id ::= String2Id ( String )    [function]
-  rule String2Id(S:String) => #parseToken("Id", S)
-
+  syntax Id ::= String2Id (String) [function, hook(STRING.string2token)]
```

Comment:
I see in the list of hooks there is a `string2id`, why use `string2token` here?
Wait, you just removed `string2token`. Can you take another look at this?

Reply:
This is the version of the module used by the ocaml and kore backends, not by the java backend